pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -178,3 +193,55 @@ func (i *deploymentClientImpl) listDeploymentsForConfig(namespace, configName st
 func (i *deploymentClientImpl) updateDeployment(namespace string, deployment *kapi.ReplicationController) (*kapi.ReplicationController, error) {
 	return i.updateDeploymentFunc(namespace, deployment)
 }
+
+// failureInspection inspects the given deployment configuration for any failure causes
+// and updates the status of the configuration accordingly
+func (c *DeploymentConfigController) failureInspection(config *deployapi.DeploymentConfig, outErr error) error {
+	if config.Details == nil {
+		config.Details = new(deployapi.DeploymentDetails)
+	}
+	// See if the provided error is non-nil and use its message as the dc message
+	if outErr != nil {
+		config.Details.Message = outErr.Error()
+		c.configClient.DeploymentConfigs(config.Namespace).Update(config)
+		return nil
+	}
+	// Look into image change triggers and find out possible deployment failures such as
+	// missing image stream tags with or without build configurations pointing at them
+	for _, trigger := range config.Triggers {
+		if trigger.Type != deployapi.DeploymentTriggerOnImageChange || trigger.ImageChangeParams == nil {
+			continue
+		}
+		name := trigger.ImageChangeParams.From.Name
+		tag := trigger.ImageChangeParams.Tag
+
+		// Check if the image stream tag pointed by the trigger exists
+		if _, err := c.istagClient.ImageStreamTags(config.Namespace).Get(name, tag); err != nil {
+			defer c.configClient.DeploymentConfigs(config.Namespace).Update(config)
+			// In case the image stream tag was not found, then it either doesn't exist or doesn't exist yet
+			// (a build configuration output points to it so it's going to be populated at some point in the
+			// future)
+			if errors.IsNotFound(err) {
+				config.Details.Message = fmt.Sprintf("The config trigger for %q will have no effect because \"%s:%s\" does not exist", deployutil.LabelForDeploymentConfig(config), name, tag)
+				bcList, err := c.bcClient.BuildConfigs(kapi.NamespaceAll).List(labels.Everything(), fields.Everything())
+				if err != nil {
+					// We don't really care about this error. The dc message will be updated
+					// with the missing tag warning anyway
+					return nil
+				}
+				for _, bc := range bcList.Items {
+					if bc.Spec.Output.To != nil && bc.Spec.Output.To.Kind == "ImageStreamTag" {
+						parts := strings.Split(bc.Spec.Output.To.Name, ":")
```

Comment:
Better do an assertion on the length of `parts` to avoid a panic if length is less than 2...